web/views.py

```
@@ -6949,3 +6951,168 @@ def delete_post(request, post_id):
+    except Exception:
+        subject_name = getattr(subject, "name", "General") if "subject" in locals() else "General"
+        return JsonResponse({"fact": get_fallback_fact(subject_name)})
+
+
+def generate_fact_for_subject(subject_name):
+    """Generate a fact for a subject using Google's Gemini API."""
+    try:
+        api_key = os.environ.get("GEMINI_API_KEY")
+        if not api_key:
+            logger = logging.getLogger(__name__)
+            logger.warning(f"Gemini API key not found, using fallback fact for {subject_name}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)